proto/span.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+package span;
+
+enum Event {
+    Unknown = 0;
+    CoprRequest = 1;
+    Scheduled = 2;
+    Snapshot = 3;
+    HandleRequest = 4;
+    HandleUnaryFixture = 5;
+    HandleChecksum = 6;
+    HandleDag = 7;
+    HandleBatchDag = 8;
+    HandleAnalyze = 9;
+    HandleCached = 10;
+};
+
+message SpanSet {
+    uint64 start_time_ns = 1;
+    uint64 cycles_per_sec = 2;
+    repeated Span spans = 3;
+}
+
+message Root {}
+message Parent { uint64 id = 1; }
+message Continue { uint64 id = 1; }
+
+message Link {
+    oneof link {
+        Root root = 1;
+        Parent parent = 2;
+        Continue continue = 3;
+    }
+}
+
+
+message Span {
+    uint64 id = 1;
+    Link link = 2;
+    uint64 begin_cycles = 3;
```

Comment:
counted by cycles instead of ns?

Reply:
We leave the conversion work to the visualization part since heavy div jobs are needed.

Reply:
The time cost of an integer div each span might not be so notable and worth to have a try

Reply:
The benchmark result leads me to make this decision. An integer div op cost more than 10ns.